torch/jit/annotations.py

```diff
@@ -244,51 +244,52 @@ def as_ann(ann):
 def try_ann_to_type(ann, loc):
     if ann is None:
         return TensorType.get()
-    elif ann is torch.Tensor:
+    if ann is torch.Tensor:
         return TensorType.get()
-    elif is_tuple(ann):
+    if is_tuple(ann):
         return TupleType([try_ann_to_type(a, loc) for a in ann.__args__])
-    elif is_list(ann):
+    if is_list(ann):
         return ListType(try_ann_to_type(ann.__args__[0], loc))
-    elif is_dict(ann):
+    if is_dict(ann):
         key = try_ann_to_type(ann.__args__[0], loc)
         value = try_ann_to_type(ann.__args__[1], loc)
         return DictType(key, value)
-    elif is_optional(ann):
+    if is_optional(ann):
         if issubclass(ann.__args__[1], type(None)):
             return OptionalType(try_ann_to_type(ann.__args__[0], loc))
         else:
             return OptionalType(try_ann_to_type(ann.__args__[1], loc))
-    elif is_rref(ann):
+    if is_rref(ann):
         return RRefType(try_ann_to_type(ann.__args__[0], loc))
-    elif is_future(ann):
+    if is_future(ann):
         return FutureType(try_ann_to_type(ann.__args__[0], loc))
-    elif ann is float:
+    if ann is float:
         return FloatType.get()
-    elif ann is int:
+    if ann is int:
         return IntType.get()
-    elif ann is str:
+    if ann is str:
         return StringType.get()
-    elif ann is bool:
+    if ann is bool:
         return BoolType.get()
-    elif ann is Any:
+    if ann is Any:
         return AnyType.get()
-    elif ann is type(None):
+    if ann is type(None):
         return NoneType.get()
-    elif inspect.isclass(ann) and hasattr(ann, "__torch_script_class__"):
-        return ClassType(_qualified_name(ann))
-    elif inspect.isclass(ann) and hasattr(ann, "__torch_script_interface__"):
+    if inspect.isclass(ann) and hasattr(ann, "__torch_script_interface__"):
         return InterfaceType(_qualified_name(ann))
-    elif ann is torch.device:
+    if ann is torch.device:
         return DeviceObjType.get()
-    else:
-        # Maybe resolve a NamedTuple to a Tuple Type
-        def fake_rcb(key):
-            return None
-        the_type = torch._C._resolve_type_from_object(ann, loc, fake_rcb)
-        if the_type is not None:
-            return the_type
-    return None
+    if inspect.isclass(ann):
+        if hasattr(ann, "__torch_script_class__"):
+            return ClassType(_qualified_name(ann))
+        if torch._jit_internal.can_compile_class(ann):
+            torch.jit._recursive_compile_class(ann, loc)
```

Comment:
Will the extra information here help at all? If anything, we should be explaining _why_ we are compiling this class by adding an error stack entry here. Otherwise eventhough we know we are compiling class X, we don't know why we are compiling it.

Reply:
the error stack should show why we are compiling class X as well